agent/ec2_meta_data.go

```diff
@@ -36,23 +36,16 @@ func (e EC2MetaData) Get() (map[string]string, error) {
 		return metaData, err
 	}
 
-	instanceId, err := c.GetMetadata("instance-id")
+	document, err := c.GetInstanceIdentityDocument()
 	if err != nil {
 		return metaData, err
 	}
-	metaData["aws:instance-id"] = string(instanceId)
 
-	instanceType, err := c.GetMetadata("instance-type")
-	if err != nil {
-		return metaData, err
-	}
-	metaData["aws:instance-type"] = string(instanceType)
-
-	amiId, err := c.GetMetadata("ami-id")
-	if err != nil {
-		return metaData, err
-	}
-	metaData["aws:ami-id"] = string(amiId)
+	metaData["aws:instance-id"] = string(document.InstanceID)
+	metaData["aws:instance-type"] = string(document.InstanceType)
+	metaData["aws:ami-id"] = string(document.ImageID)
+	metaData["aws:availability-zone"] = string(document.AvailabilityZone)
+	metaData["aws:region"] = string(document.Region)
```

Comment:
why are we casting all of these fields to strings? AFAICT they're [already strings](https://github.com/aws/aws-sdk-go/blob/main/aws/ec2metadata/api.go#L234-L250)

Reply:
Pure cargo culting on my part. Remove the cast if you like?

Reply:
yeah, uncast them

Reply:
Done!